### ISSUE TITLE
Fapi CB returns shall be const [API-break]

### ISSUE THREAD

**Issue by AndreasFuchsSIT** [tpm2-software/tpm2-tss#1709]
There are two callbacks in FAPI that return pointers to some data to the FAPI library for use.
The FAPI library will neither alter nor free these data.
Thus, the return parameters should be const.

https://github.com/tpm2-software/tpm2-tss/blob/ec48d5af2e46b1fcda49259fe441c50891fd5481/include/tss2/tss2_fapi.h#L658
https://github.com/tpm2-software/tpm2-tss/blob/ec48d5af2e46b1fcda49259fe441c50891fd5481/include/tss2/tss2_fapi.h#L687

**Comment by AndreasFuchsSIT** [comment]
Also the returned value must not be free'd by FAPI:
https://github.com/tpm2-software/tpm2-tss/blob/20c5ab3a11d2d8548a99761fbcda0116befc3462/src/tss2-fapi/fapi_util.c#L347
This second part is [backport]

**Comment by AndreasFuchsSIT** [comment]
Ok, so I did a little more thinking: IMHO we should remove the fapi-context from the CBs since the fapi-context is locked at this point anyways and there is nothing to be done at that moment...

Instead, we should add the ObjectPath that we are authenticating. That way, non-user interacting software has a much easier time working with fapi.
As such, I'm proposing the following:
```
@@ -1,11 +1,11 @@
 typedef TSS2_RC (*Fapi_CB_Auth)(
-    FAPI_CONTEXT   *context,
+    char     const *objectPath,
     char     const *description,
-    char          **auth,
+    char    const **auth,
     void           *userData);
 
 typedef TSS2_RC (*Fapi_CB_Branch)(
-    FAPI_CONTEXT   *context,
+    char     const *objectPath,
     char     const *description,
     char    const **branchNames,
     size_t          numBranches,
@@ -13,19 +13,19 @@
     void           *userData);
 
 typedef TSS2_RC (*Fapi_CB_Sign)(
-    FAPI_CONTEXT   *context,
+    char     const *objectPath,
     char     const *description,
     char     const *publicKey,
     char     const *publicKeyHint,
     uint32_t        hashAlg,
     uint8_t  const *dataToSign,
     size_t          dataToSignSize,
-    uint8_t       **signature,
+    uint8_t const **signature,
     size_t         *signatureSize,
     void           *userData);
 
 typedef TSS2_RC (*Fapi_CB_PolicyAction)(
-    FAPI_CONTEXT   *context,
+    char     const *objectPath,
     char     const *action,
     void           *userData);
```

I'd like to get some feedback on this by you, @williamcroberts @tstruk @JuergenReppSIT 
This would break API but we are moving towards 3.0 anyways and I think it would be worth it...
And for existing software we can have some very easy ifdefs...

**Comment by JuergenReppSIT** [comment]
I think that's much better than the old version.

**Comment by williamcroberts** [comment]
+1 make 3.0+

**Comment by AndreasFuchsSIT** [comment]
Fixed via #1719
